### PR TITLE
Add new extra large size (.modal-xl) for .modal-dialog

### PR DESCRIPTION
**Pull request by malkomich**

### CODE REVIEW

File: docs/components/modal.md [hunk 1]
```diff
@@ -189,6 +189,17 @@ Modals have two optional sizes, available via modifier classes to be placed on a
 </div>
 
 {% highlight html %}
+<!-- Extra large modal -->
+<button class="btn btn-primary" data-toggle="modal" data-target=".bd-example-modal-xl">Extra large modal</button>
```
Comment (cvrebert) [inline docs/components/modal.md]: Missing another button for this around line 187

**Comment by cvrebert** [comment]
You also need to update line 184 to reflect the addition of your new size. Currently it says "Modals have **two** optional sizes".

**Comment by mdo** [comment]
Why add another one?

**Comment by malkomich** [comment]
@mdo Because it was useful for me to display large tables and forms in a modal container, so I guess someone could enjoy this small enhancement.

And also it doesn't break anything..so why not? :smile:

**Comment by ASHFAQPATWARI** [comment]
This is nice to have option.

**Comment by bogdaniel** [comment]
actully i'm working with some modals now and was thinking i should increase the size.

File: scss/_modal.scss [hunk 2]
```diff
@@ -140,3 +140,7 @@
 @include media-breakpoint-up(lg) {
   .modal-lg { max-width: $modal-lg; }
```
Comment (yukihirokawauchiya) [inline scss/_modal.scss]: Maybe we can add a line below to here for smooth transition between md and lg break points.
  .modal-xl { max-width: $modal-lg; }

**Review by TheSachin — APPROVED** [review]

**Comment by chroda** [comment]
This is so small and non-disruptive enhancement, guess is taken to long to get aproved the PR

**Comment by mdo** [comment]
Update based on this PR coming in #26821.

**Pull request closed without merging**